tempest/api/workloadmgr/cli/test_wlm_cli.py

```diff
@@ -276,3 +276,221 @@ def test_05_create_workload_setting_with_invalid_values(self):
+                reporting.add_test_step(
+                    "Verify workload setting get-hidden after updation", tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify workload setting get-hidden after updation", tvaultconf.FAIL)
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.set_test_script_status(tvaultconf.FAIL)
+        finally:
+            reporting.test_case_to_write()
+
+    @decorators.attr(type='workloadmgr_cli')
+    def test_07_update_workload_setting_with_invalid_values(self):
+        reporting.add_test_script(str(__name__) + "_update_workload_setting_cli_with_invalid_values")
+        try:
+            cli_error_string = tvaultconf.wl_setting_update_cli_error_string
+            wl_setting_update = command_argument_string.workload_setting_update + " " + workload_setting_name + \
+                                " value45 --description --type set2"
+            LOG.debug("Workload setting update command with invalid values: {}".format(wl_setting_update))
+            rc = cli_parser.cli_returncode(wl_setting_update)
+            if rc == 0:
+                reporting.add_test_step(
+                    "Execute workload setting-update with invalid values", tvaultconf.FAIL)
+                raise Exception("workload setting-update command did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload setting-update with invalid values", tvaultconf.PASS)
+                LOG.debug("workload setting-update command executed correctly")
+
+            cli_err = cli_parser.cli_error(wl_setting_update)
+            LOG.debug("cli error: {}".format(cli_err))
+            if (cli_err and cli_error_string in cli_err):
+                reporting.add_test_step(
+                    "Verify correct error thrown for invalid parameter", tvaultconf.PASS)
+                LOG.debug("Command executed correctly")
+            else:
+                reporting.add_test_step(
+                    "Verify correct error thrown for invalid parameter", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            # Cleanup
+            # Delete workload setting
+            wl_setting_delete = command_argument_string.workload_setting_delete + \
+                                workload_setting_name
+            LOG.debug("Workload setting delete command: {}".format(wl_setting_delete))
+            rc = cli_parser.cli_returncode(wl_setting_delete)
+            time.sleep(10)
+            wc = query_data.get_created_workload_setting(workload_setting_name)
+            LOG.debug("Workload setting status: " + str(wc))
+
+            if wc:
+                reporting.add_test_step(
+                    "workload setting deletion failed", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
```

Comment:
Same as above statement. 
Instead of raising an exception, set_test_script_status() is called.
Please confirm.

Reply:
In case of failures, in this case, we mark the step as fail and continue execution (This does not fail entire test execution)